### ISSUE TITLE
Convert to astropy.time

### ISSUE THREAD

**Issue by jtniehof** [spacepy/spacepy#52]
This is a big "maybe". It may be worth replacing ticktock or rebasing it on astropy.time (http://docs.astropy.org/en/stable/time/). Open for discussion here....

**Comment by drsteve** [comment]
It looks like the underlying library behind that has very similar
functionality to LANLGeoMag's time stuff. I do note that leap seconds are
hardcoded in their implementation, so can't be trivially updated by the
user.

I'm happy to compare/contrast, but if we're going to wrap C I think we
should take a look at LGM too.

**Comment by jtniehof** [comment]
I absolutely would love to switch over to LGM for coordinates, and if we get time for it reasonably, so much the better. AstroPy time obviously has more people using it but hardcoded leapseconds are a huge bummer.

**Comment by drsteve** [comment]
Remember that an overhaul of coordinates is on my list, and I'm looking at
LGM there. It might be easiest to pull out the time and coordinates
together from LGM and wrap that. We should talk when I get back.

**Comment by jtniehof** [comment]
Yes, we should totally do that...I'm talking to a plasmapy guy here about coordinates, too.

**Comment by jtniehof** [comment]
Hmmm, I wonder if it makes sense to steal the AstroPy time API so people can use either one with similar code.

**Comment by jtniehof** [comment]
I've been told astropy has a coordinate transformation framework into which you can register other conversion routines. Might be worth supporting.

**Comment by ayshih** [comment]
SunPy makes use of Astropy's coordinates framework, which we extend with coordinate frames relevant to solar physics.  Each coordinate frame can be transformed to any other coordinate frame in either SunPy or Astropy.  Here's an illustrative transformation graph:
![transform_whatsnew](https://user-images.githubusercontent.com/991759/76759908-02864a80-6763-11ea-88d6-f758aeceac64.png)
